### PR TITLE
Extend Connection with remoteAddress and eventLoop

### LINKED ISSUE #93: [MINOR FEATURE] Enable access to the current `EventLoop` trough the `@Environment` property wrapper
**Is your feature request related to a problem? Please describe.**
Some `Handler`s or `Guard`s rely on asynchronous work that has to be performed on an `EventLoop`. We should provide the Apodini developer access to the current `EventLoop` the request is handled on to e.g. execute tasks on the `EventLoop` or create Futures/Promises.

**Propose a solution**
I would suggest to add an other key path to the `@Environment` property wrapper to access the event loop:
```swift
struct MyHandler: Handler {
    @Apodini.Environment(\.eventLoop) var eventLoop: EventLoop
    
    func handle() -> EventLoopFuture<String> {
        // The `EvenLoop` can be used in the handle method
    }
}
```

### LINKED ISSUE #123: [MINOR FEATURE] @Parameter for remote address
**Is your feature request related to a problem? Please describe.**
I would like to use the IP address of a request in my `Handler`.

**Propose a solution**
Add a type like `@AddressParameter`

**Additional context**
`Vapor.Request.remoteAddress` exposes a `SocketAddress` for us to use.

### PR DESCRIPTION
**Pull request by tgymnich**
# Extend Connection with remoteAddress and eventLoop

## :recycle: Current situation
It is not possible to retrieve any metadata about the request besides the connection state.

## :bulb: Proposed solution
Add `remoteAddress` and `eventLoop` to `Connection`.

### Problem that is solved
More request metadata is available.

fixes #93
fixes #123

### Reviewer Nudging
Check out the changes to `ExporterRequest` for your exporter.

### CODE REVIEW

**Comment by tgymnich** [comment]
@theMomax I could use your help to add this to WebSockets as well.

**Comment by theMomax** [comment]
@tgymnich I added support for WebSocket. I'll leave the review to the others.

**Comment by tgymnich** [comment]
@theMomax thanks a lot 👍

File: Sources/Apodini/Semantic Model Builder/InterfaceExporter/Validator.swift [hunk 1]
```diff
@@ -109,8 +109,17 @@ private class EndpointValidator<I: InterfaceExporter, H: Handler>: Validator {
...
+        if let requestWithRemote = request as? WithRemote {
+            return ValidatedRequest(for: exporter,
+                                    with: request,
+                                    using: output,
+                                    on: endpoint,
+                                    running: eventLoop,
+                                    remoteAddress: requestWithRemote.remoteAddress)
+        } else {
+            return ValidatedRequest(for: exporter, with: request, using: output, on: endpoint, running: eventLoop, remoteAddress: nil)
+        }
```
Comment (nityanandaz) [inline Sources/Apodini/Semantic Model Builder/InterfaceExporter/Validator.swift]:
```suggestion
        let remoteAddress = (request as? WithRemote)?.remoteAddress
        return ValidatedRequest(for: exporter, with: request, using: output, on: endpoint, running: eventLoop, remoteAddress: remoteAddress)
```

Just a little bit shorter 🙂

**Review by nityanandaz — APPROVED** [review]
The code looks good to me 👍

File: Sources/Apodini/Semantic Model Builder/InterfaceExporter/Validator.swift (hunk 1 above)
Comment (tgymnich) [reply Sources/Apodini/Semantic Model Builder/InterfaceExporter/Validator.swift:122]: I think swift lint won't allow this 😄

**Pull request merged**